client/src/Components/PrivateRoute/PrivateRoute.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from "react";
+import { withRouter, Route, Redirect } from "react-router-dom";
+import { connect } from "react-redux";
+
+import HomeView from "../HomeView/HomeView";
+
+class PrivateRoute extends Component {
+  render() {
+    let protectedComponent = props => {
+      let { userRole, userID, component, tickets, path } = props;
```

Comment:
you can just change this to `this.props` and you'll be just fine without the wrapping function